tools/elfdeps.c

```diff
@@ -90,11 +93,106 @@ static const char *mkmarker(GElf_Ehdr *ehdr)
 	    /* alpha doesn't traditionally have 64bit markers */
 	    break;
 	default:
-	    marker = "(64bit)";
+	    classmarker = "(64bit)";
 	    break;
 	}
     }
-    return marker;
+    return classmarker;
+}
+
+static const char *mkarchmarker(GElf_Ehdr *ehdr)
+{
+    char *archmarker = NULL;
+    const char *elf_machine = NULL;
+    const char *elf_endian = NULL;
+    const char *elf_bitsize = NULL;
+
+    /* First the machine type... */
+    switch (ehdr->e_machine) {
+    case EM_SPARC:
+	elf_machine = "sparc";
+	break;
+    case EM_386:
+    case EM_X86_64:
+	elf_machine = "x86";
+	break;
+    case EM_MIPS:
+	elf_machine = "mips";
+	break;
+    case EM_PPC:
+    case EM_PPC64:
+	elf_machine = "ppc";
+	break;
+    case EM_S390:
+	elf_machine = "s390";
+	break;
+    case EM_ARM:
+	if ((ehdr->e_flags | EF_ARM_ABI_FLOAT_HARD) == EF_ARM_ABI_FLOAT_HARD)
+		elf_machine = "armhfp";
+	if ((ehdr->e_flags | EF_ARM_ABI_FLOAT_SOFT) == EF_ARM_ABI_FLOAT_SOFT)
+		elf_machine = "armsfp";
```

Comment:
Those two if statements look fishy. I expected something like `(ehdr->e_flags & ~EF_ARM_ABI_FLOAT_HARD) == EF_ARM_ABI_FLOAT_HARD`

Reply:
Oops, fixed!

Reply:
Erk, nope, that's no bueno:

```
tools/elfdeps.c:130:47: error: bitwise comparison always evaluates to false [-Werror=tautological-compare]
  130 |  if ((ehdr->e_flags & ~EF_ARM_ABI_FLOAT_HARD) == EF_ARM_ABI_FLOAT_HARD)
      |                                               ^~
tools/elfdeps.c:132:47: error: bitwise comparison always evaluates to false [-Werror=tautological-compare]
  132 |  if ((ehdr->e_flags & ~EF_ARM_ABI_FLOAT_SOFT) == EF_ARM_ABI_FLOAT_SOFT)
      |                                               ^~
```

Reply:
Oops, did I write `& ~EF_ARM_ABI_FLOAT_HARD`? I meant `& EF_ARM_ABI_FLOAT_HARD`, of course. Sorry. Good thing that this triggers a gcc warning...

Reply:
Fixed.